docs/intro-to-coalton.md

```diff
@@ -4,15 +4,63 @@ Coalton is a statically typed language that is embedded in, and compiles to, Com
+This is a convenience package that is defined purely for your interactive and experimental use. (No software should ever depend on it, just like `#:cl-user`.)
+
+When making a new project, you will want to define a new package and establish your standard library imports. *Do not `:use` the `#:cl` or `#:common-lisp` packages!* Instead, you'll want to, at minimum, `:use` both the `#:coalton` package (for core language features) and `#:coalton-prelude` (for extremely common standard library definitions):
+
+```lisp
+(defpackage #:my-package
+  (:use #:coalton #:coalton-prelude))
+```
+
+The `#:coalton-prelude` package does not contain any functionality that isn't also found elsewhere in the standard library, and hence, it is a convenience.
+
+In idiomatic usage, this minimal package definition would be far too limiting to be useful. For instance, if we wanted to use the standard library function `strip-prefix` (which strips a prefix off of a string), we would have to type `coalton-library/string:strip-prefix` in our program. Instead, we can create a local nickname for this package:
+
+```lisp
+(defpackage #:my-package
+  (:use #:coalton)
+  (:local-nicknames (#:str #:coalton-library/string)))
+```
+
+This way, we can now just type `str:strip-prefix`. Any time we make use of a new function, either from the Coalton standard library or from a third-party library, we might add nicknames that function's package to our `:local-nicknames` list. For example:
+
+```lisp
+(defpackage #:my-package
+  (:use #:coalton)
+  (:local-nicknames (#:str  #:coalton-library/string)
+                    (#:vec  #:coalton-library/vector)
+                    (#:math #:coalton-library/math)))
+```
+
+**We do not recommend `:use`ing any other package besides `#:coalton` and `#:coalton-prelude`.** Historically, in Common Lisp, this has caused backwards-compatibility issues when the packages themselves introduce new exported symbols. Moreover, unlike Common Lisp, Coalton follows a pattern of using the same symbol name for similarly defined functions of different packages. For example, both the string package and the vector package have a symbol named `length`, i.e., `str:length` and `vec:length` both exist.
+
+
+## Program Structure
+
+After creating your package `#:my-package`, you must switch to it with:
+
+```lisp
+(named-readtables:in-readtable coalton:coalton)
+
+(in-package #:my-package)
+```
+
+The `named-readtables` liine is optional, but (assuming you've included it as a dependency to your system definitition) you will have vastly better error messages. Using Coalton's reader allow for compiler errors to accurately refer to source code and provide correct line numbers.
+
+Alternatively, you will have switched to `#:coalton-user` in the REPL. Remember that Coalton packages do *not* `:use` the `#:common-lisp`/`#:cl` package, so you must prepend Common Lisp symbols with `cl:` if you need them.
```

Comment:
i think it's too premature to declare writing Lisp in Coalton as rare, as everybody who has written Coalton has needed to do it extensively so far

Reply:
Yeah, true. I'm looking for some way to prevent people from trying to call `cl:print-line` in Coalton code without a `lisp` form and wondering why it doesn't work, when the docs call out that you can refer to CL operators using the `cl:` prefix.